website/for_teachers.py

```diff
@@ -1113,7 +1112,10 @@ def update_adventure(self, user):
         for class_id in body["classes"]:
             if class_id != '0':
                 for level in body["levels"]:
-                    self.add_adventure_to_class_level(user, class_id, body["id"], level, False)
+                    if level not in current_levels and class_id in current_classes:
+                        self.add_adventure_to_class_level(user, class_id, body["id"], level, False)
+                    elif class_id not in current_classes:
+                        self.add_adventure_to_class_level(user, class_id, body["id"], level, False)
```

Comment:
why are they in if-elif statement? The body is the same!
```suggestion
                    if class_id not in current_classes or (level not in current_levels and class_id in current_classes):
                        self.add_adventure_to_class_level(user, class_id, body["id"], level, False)
```